Out of date `deasync` package causing blocking build error

### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When running `pnpm install` to initialize my dev environment, I immediately receive the following error:
````
│   CXX(target) Release/obj.target/deasync/src/deasync.o
│ ../src/deasync.cc:3:10: fatal error: napi.h: No such file or directory
│     3 | #include <napi.h>
│       |          ^~~~~~~~
│ compilation terminated.
│ make: *** [deasync.target.mk:111: Release/obj.target/deasync/src/deasync.o] Error 1
│ make: Leaving directory '/home/joelt/Local Sites/mothrad/app/public/wp-content/plugins/woocommerce/node_modules/.pnpm/deasync@0.1.21/node_modules/deasync/build'
│ gyp ERR! build error 
│ gyp ERR! stack Error: `make` failed with exit code: 2
│ gyp ERR! stack     at ChildProcess.onExit (/home/joelt/Local Sites/mothrad/app/public/wp-content/plugins/woocommerce/node_modules/.pnpm/node-gyp@8.4.1/node_modules/node-gyp/lib/build.js:194:23)
│ gyp ERR! stack     at ChildProcess.emit (node:events:390:28)
│ gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
│ gyp ERR! System Linux 5.16.15-76051615-generic
│ gyp ERR! command "/home/joelt/.nvm/versions/node/v16.13.1/bin/node" "/home/joelt/Local Sites/mothrad/app/public/wp-content/plugins/woocommerce/node_modules/.pnpm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
│ gyp ERR! cwd /home/joelt/Local Sites/mothrad/app/public/wp-content/plugins/woocommerce/node_modules/.pnpm/deasync@0.1.21/node_modules/deasync
│ gyp ERR! node -v v16.13.1
│ gyp ERR! node-gyp -v v8.4.1
│ gyp ERR! not ok 
│ Build failed
````

This originated from the `deasync` package, and while searching for preexisting issues I came across this [fairly similar one](https://href.li/?https://github.com/woocommerce/woocommerce/issues/31882). I do have spaces in my path, so perhaps that is related, although that would likely impact anyone using Flywheel’s Local application. I’ve also seen [similar issues](https://href.li/?https://github.com/nodejs/node-addon-api/pull/766) reportedly stemming from the `node-addon-api` dependency.

**Note that I've been able to work around this simply by updating the `deasync` package to the latest patch version `0.1.26`**

### Expected behavior

Running `pnpm install` installs all required packages with no error.

### Actual behavior

The above error is encountered, and the command fails.

### Steps to reproduce

I'm not entirely certain what about my environment is causing this issue, but it likely is related to the fact that I'm on Linux, and perhaps that I have spaces in my directory structure, as mentioned in [the other issue](https://github.com/woocommerce/woocommerce/issues/31882).

### WordPress Environment

The WooCommerce plugin can't be build, so no system status report, but this is the latest `trunk` with WP 5.9.3.

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

--- a/plugins/woocommerce/changelog/update-32509_deasync_issue
+++ b/plugins/woocommerce/changelog/update-32509_deasync_issue
@@ -1,5 +1,5 @@
 Significance: patch
 Type: update
-Comment: Simply updating a patch version of the deasync package
+Updating deasync package to resolve install script issue with Linux
 
 

--- a/plugins/woocommerce/changelog/update-32509_deasync_issue
+++ b/plugins/woocommerce/changelog/update-32509_deasync_issue
@@ -1,0 +1,5 @@
+Significance: patch
+Type: update
+Comment: Simply updating a patch version of the deasync package
+
+

--- a/plugins/woocommerce/package.json
+++ b/plugins/woocommerce/package.json
@@ -55,7 +55,7 @@
     "chai-as-promised": "7.1.1",
     "config": "3.3.3",
     "cross-env": "6.0.3",
-    "deasync": "0.1.21",
+    "deasync": "0.1.26",
     "eslint": "6.8.0",
     "eslint-config-wpcalypso": "5.0.0",
     "eslint-plugin-jest": "23.20.0",

